okonomiyaki/file_formats/_egg_info.py

```diff
@@ -142,6 +144,12 @@ def parse_rawspec(spec_string):
             res[key] = spec[key]
         except KeyError:
             raise InvalidMetadataField(key, InvalidMetadataField.undefined)
+
+    for k, v in res.items():
+        if isinstance(v, six.binary_type):
+            res[k] = decode_if_needed(v)
```

Comment:
Why is it called `decode_if_needed` if you have to manually check the value first? Can this just be a comprehension rather than a loop that modifies the dict while you're iterating over it? I know it's safe in this case, but why do it?